iib/web/models.py

```diff
@@ -390,6 +399,64 @@ class Request(db.Model):
         'polymorphic_on': 'type',
     }
 
+    def from_json_replacement_even_shorter(
+        self,
+        payload: UnionPydanticRequestType,
+        batch: Optional[Batch] = None,
+        batch_allowed: Optional[bool] = False,
+        build_tags_allowed: Optional[bool] = False,
+    ):
+        """
+        Handle JSON requests for the builds/* API endpoint.
+
+        :param UnionPydanticRequestType payload: the Pydantic model representing the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = payload.get_json_for_request()
+
+        keys_to_check = payload.get_keys_to_check_in_db()
+        for key in keys_to_check:
+            if key in [
+                'binary_image',
+                'fbc_fragment',
+                'from_index',
+                'from_bundle_image',
+                'source_from_index',
+                'target_index',
+                'parent_bundle_image',
+            ]:
+                request_kwargs[key] = Image.get_or_create(pull_specification=request_kwargs[key])
+
+            elif key in ["bundles", "deprecation_list"]:
+                images = request_kwargs.get(key, [])
+                request_kwargs[key] = [
+                    Image.get_or_create(pull_specification=image) for image in images
+                ]
+
+            elif key == ["operators"]:
+                request_kwargs['operators'] = [Operator.get_or_create(name=item) for item in request_kwargs["operators"]]
+            else:
+                raise ValidationError(f"Unexpected key: {key} during from_json() method.")
+
+        # current_user.is_authenticated is only ever False when auth is disabled
+        if current_user.is_authenticated:
+            request_kwargs['user'] = current_user
+
+        # Add the request to a new batch
+        if batch_allowed:
+            batch = batch or Batch()
+            db.session.add(batch)
+            request_kwargs['batch'] = batch
+
+        request = self(**request_kwargs)
+
+        if build_tags_allowed:
+            for bt in payload.build_tags:
+                request.add_build_tag(bt)
+
+        request.add_state('in_progress', 'The request was initiated')
+        return request
+
```

Comment:
All code in from_json() methods could be reduced to this one function. 

Line 451 is giving me a warning that the method cannot call itself. I intended to allow other Request classes to inherit this 'from_json_replacement_even_shorter' method via the Request parent class.